pkg/apis/coordination/validation/validation.go

```diff
@@ -115,25 +117,33 @@ func ValidateLeaseCandidateSpec(spec *coordination.LeaseCandidateSpec, fldPath *
 	}
 	if !bv.EQ(ev) && bv.LT(ev) {
 		fld := fldPath.Child("binaryVersion")
-		allErrs = append(allErrs, field.Invalid(fld, spec.BinaryVersion, "must be greater than or equal to `emulationVersion"))
+		allErrs = append(allErrs, field.Invalid(fld, spec.BinaryVersion, "must be greater than or equal to `emulationVersion`"))
 	}
 
-	if spec.CandidateDurationSeconds <= 0 {
-		fld := fldPath.Child("candidateDurationSeconds")
-		allErrs = append(allErrs, field.Invalid(fld, spec.CandidateDurationSeconds, "must be greater than 0"))
+	if spec.CandidateDurationSeconds == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("candidateDurationSeconds"), ""))
+	} else if spec.CandidateDurationSeconds < 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("candidateDurationSeconds"), spec.CandidateDurationSeconds, "must be greater than 0"))
 	}
 
+	strategiesMap := make(map[coordination.CoordinatedLeaseStrategy]bool)
+
 	if len(spec.PreferredStrategies) > 0 {
 		for i, strategy := range spec.PreferredStrategies {
 			fld := fldPath.Child("preferredStrategies").Index(i)
+			if _, ok := strategiesMap[strategy]; ok {
+				allErrs = append(allErrs, field.Duplicate(fld, strategy))
+			} else {
+				strategiesMap[strategy] = true
+			}
 
 			if strategy == coordination.OldestEmulationVersion {
 				zeroVersion := semver.Version{}
 				if bv.EQ(zeroVersion) {
-					allErrs = append(allErrs, field.Required(fldPath.Child("binaryVersion"), "must be set if strategy is OldestEmulationVersion"))
+					allErrs = append(allErrs, field.Required(fldPath.Child("binaryVersion"), "must be specified if `strategy` is 'OldestEmulationVersion'"))
 				}
 				if ev.EQ(zeroVersion) {
-					allErrs = append(allErrs, field.Required(fldPath.Child("emulationVersion"), "must be set if strategy is OldestEmulationVersion"))
+					allErrs = append(allErrs, field.Required(fldPath.Child("emulationVersion"), "must be specified `strategy` is 'OldestEmulationVersion'"))
```

Comment:
missing an "if"  (actually should be "when")

Reply:
fixed